.teamcity/src/main/kotlin/configurations/Gradleception.kt

```diff
@@ -20,25 +26,67 @@ class Gradleception(model: CIBuildModel, stage: Stage) : BaseGradleBuildType(sta
+
+     */
+    val dogfoodTimestamp1 = "20210101010101+0000"
+    val dogfoodTimestamp2 = "20210202020202+0000"
     val buildScanTagForType = buildScanTag("Gradleception")
     val defaultParameters = (buildToolGradleParameters() + listOf(buildScanTagForType) + "-Porg.gradle.java.installations.auto-download=false").joinToString(separator = " ")
 
-    applyDefaults(model, this, ":distributions-full:install", notQuick = true, extraParameters = "-Pgradle_installPath=dogfood-first $buildScanTagForType", extraSteps = {
-        localGradle {
-            name = "BUILD_WITH_BUILT_GRADLE"
-            tasks = "clean :distributions-full:install"
-            gradleHome = "%teamcity.build.checkoutDir%/dogfood-first"
-            gradleParams = "-Pgradle_installPath=dogfood-second -PignoreIncomingBuildReceipt=true $defaultParameters"
-        }
-        localGradle {
-            name = "QUICKCHECK_WITH_GRADLE_BUILT_BY_GRADLE"
-            tasks = "clean sanityCheck test"
-            gradleHome = "%teamcity.build.checkoutDir%/dogfood-second"
-            gradleParams = defaultParameters
-        }
-    })
+    applyDefaults(
+        model,
+        this,
+        ":distributions-full:install",
+        notQuick = true,
+        extraParameters = "-Pgradle_installPath=dogfood-first-for-hash -PignoreIncomingBuildReceipt=true -PbuildTimestamp=$dogfoodTimestamp1 $buildScanTagForType",
+        extraSteps = {
+            calculateMd5AndSetEnvStep("CALCULATE_MD5_VERSION_FOR_FIRST_DOGFOODING_DISTRIBUTION", "%teamcity.build.checkoutDir%/dogfood-first-for-hash")
+            gradleWrapper {
+                name = "BUILD_GRADLE_DISTRIBUTION"
+                tasks = "clean :distributions-full:install"
+                gradleParams = "-Pgradle_installPath=dogfood-first -PignoreIncomingBuildReceipt=true -PbuildTimestamp=$dogfoodTimestamp1 $defaultParameters"
+            }
+
+            calculateMd5AndSetEnvStep("CALCULATE_MD5_VERSION_FOR_SECOND_DOGFOODING_DISTRIBUTION", "%teamcity.build.checkoutDir%/dogfood-first")
+            localGradle {
+                name = "BUILD_WITH_BUILT_GRADLE"
+                tasks = "clean :distributions-full:install"
+                gradleHome = "%teamcity.build.checkoutDir%/dogfood-first"
+                gradleParams = "-Pgradle_installPath=dogfood-second -PignoreIncomingBuildReceipt=true -PbuildTimestamp=$dogfoodTimestamp2 $defaultParameters"
+            }
+
+            localGradle {
+                name = "QUICKCHECK_WITH_GRADLE_BUILT_BY_GRADLE"
+                tasks = "clean sanityCheck test"
+                gradleHome = "%teamcity.build.checkoutDir%/dogfood-second"
+                gradleParams = defaultParameters
+            }
+        })
 })
 
+fun BuildSteps.calculateMd5AndSetEnvStep(stepName: String, dir: String) {
+    script {
+        name = stepName
+        scriptContent = """
+                set -x
+                MD5=`find $dir -type f | sort | xargs md5sum | md5sum | awk '{ print ${'$'}1 }'`
```

Comment:
When doing the `xargs md5sum`, doesn't that print out absolute paths for the files which are about to be hashed? Should we somehow make these relative?

Reply:
Good point, Now I use `find . -type f` to normalize the paths.